runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBoxOverlay.java

```diff
@@ -96,11 +96,34 @@ public Dimension render(Graphics2D graphics)
 
 		final XpSnapshotSingle snapshot = plugin.getSkillSnapshot(skill);
 
+		final String leftStr;
+		final int rightNum;
+
+		switch (config.onScreenDisplayMode())
+		{
+			case ACTIONS_DONE:
+				leftStr = snapshot.getActionType().getLabel() + " Done";
+				rightNum = snapshot.getActionsInSession();
+				break;
+			case ACTIONS_LEFT:
+				leftStr = snapshot.getActionType().getLabel() + " Left";
+				rightNum = snapshot.getActionsRemainingToGoal();
+				break;
+			case XP_LEFT:
+				leftStr = config.onScreenDisplayMode().toString();
+				rightNum = snapshot.getXpRemainingToGoal();
+				break;
+			case XP_GAINED:
+			default:
+				leftStr = config.onScreenDisplayMode().toString();
+				rightNum = snapshot.getXpGainedInSession();
+				break;
+		}
+
 		final LineComponent xpLine = LineComponent.builder()
-				.left(config.displayXpLeftOnScreen() ? "XP Left:" : "XP Gained:")
-				.right(StackFormatter.quantityToRSDecimalStack(config.displayXpLeftOnScreen()
-					? snapshot.getXpRemainingToGoal() : snapshot.getXpGainedInSession()))
-				.build();
+			.left(leftStr + ":")
+			.right(StackFormatter.quantityToRSDecimalStack(rightNum))
```

Comment:
I also just noticed that I forgot to make it use the more accurate stack representation, maybe you can change it to do that here and for the XP/Hour below. ``StackFormatter.quantityToRSDecimalStack(..., true)``